### PR TITLE
Create User Profile and Refactor Backend

### PR DESCRIPTION
**Pull request by jadenpadua**
Hey Guys this is a pretty big PR, if you can try to run this on your own computer and let me know if you run into any trouble with anything. Just a heads up you will need the jwtsecret in your .env file so ask me for it and remember to npm install in root directory. Try Testing all the routes I made on PostMan if you can, if you still need help just lmk and I can meet up with you and teach you

### CODE REVIEW

**Comment by Froggerson** [comment]
pls help

**Comment by Froggerson** [comment]
I am not sure what to do.

**Comment by minecx** [comment]
I'm not part of the web team and I have no idea why @jadenpadua has been requesting my review. However I'm willing to check this out. Would you mind sending me the .env file? Thanks

**Comment by PabS729** [comment]
I cannot do npm install after pulling your branch, is there anything wrong with the packages?

**Comment by PabS729** [comment]
Update: here is the error I got
Screen Shot 2019-11-25 at 7 58 26 PM

**Comment by bryanyuan1** [comment]
@PabS729 For error installing node-pre-gyp, the problem is that this would like to have compile error with node version >= 12. So please downgrade your node version to probably 10.

**Comment by djuanit0x** [comment]
Hi @jadenpadua . I think there is a syntax error on your PR. See this [line](https://github.com/Harvesthru-Inc/Harvesthru-Web/blob/user-profile-routing/server.js#L44)

**Comment by djuanit0x** [comment]
I think you also need to remove the arrows (e.g.,` >>> and ====`) in `package.json` and `package.lock.json` due to the merge conflicts before merging it to `master` . I can't run the server because of that.

**Comment by jadenpadua** [comment]
@djuanit0x soudns good tand thank you, sorry for the delay I will repush the fixed backend sometime by tuesday. Just have to deal with some last merge copnflicts

File: package.json [hunk 1]
```diff
@@ -12,14 +12,25 @@
   "author": "",
   "license": "ISC",
   "dependencies": {
+<<<<<<< HEAD
```
Comment (hwanggit) [inline package.json]: MERGE Conflicts that need to be resolved

File: server.js [hunk 2]
```diff
@@ -31,21 +30,19 @@ mongoose
...
-// parse request body as json
+// Parse request body as json
 app.use(express.json());
+// Custom Routing for our app here
+app.use("/api/user", user);
 
-// using custom route
-app.use("/api/users", users);
 app.use("/api/auth", auth);
+app.use('/api/profile', require('./routes/profile'));
```
Comment (hwanggit) [inline server.js]: Put the require function at the top of the file like the others

File: server.js [hunk 3]
```diff
@@ -31,21 +30,19 @@ mongoose
...
 
-// render index for all other paths
 app.get("*", (req, res) => {
-  res.sendFile(path.join(__dirname, "client", "build", "index.html"));
-});
-
+// Listen on port 8000
```
Comment (hwanggit) [inline server.js]: I believe this is an unclosed { bracket

File: models/user.js [hunk 4]
```diff
@@ -1,49 +1,27 @@
...
-      maxlength: 1024
-    }
-  })
-);
+const UserSchema = new mongoose.Schema({
+  name: {
+    type: String,
+    required: true
+  },
+  email: {
```
Comment (hwanggit) [inline models/user.js]: I think we should keep the character length restraints

Comment (hwanggit) [reply models/user.js]: for name, and email only

File: package.json (hunk 1 above)
Comment (jadenpadua) [reply package.json]: resolved

File: server.js (hunk 2 above)
Comment (jadenpadua) [reply server.js]: resolved

File: server.js (hunk 3 above)
Comment (jadenpadua) [reply server.js]: resolved

File: models/user.js (hunk 4 above)
Comment (jadenpadua) [reply models/user.js]: resolved, name restraint is 64 max and email restraint is 320 max

**Pull request merged**